services/form-instances.ts

```diff
@@ -107,8 +117,6 @@ export const deleteFormInstance = async (
     throw new HttpError('Instance not found', 404);
   }
 
-  await formRepo.deleteFormInstance(form.formTtl, instanceUri);
-
-  // TODO at this stage inverse relations are kept intact even if the object gets deleted.
-  // Would be better to replace this relation with a tombstone relation.
+  const instanceTypes = await getInstanceTypes(form.formTtl, instanceUri);
```

Comment:
Personally I would fetch these instanceTypes in the deleteFormInstances, it seems like the logic belongs there more than here and you don't have access to more objects here that you would need. But that is more of a personal preference.